### PR TITLE
feat: new insights tab layout for list feature 

### LINKED ISSUE #1666: List Creation Flow - New List Page Layout
### Type of feature

🍕 Feature

### Current behavior

N/A

## [Figma File](https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/Design-Lab?type=design&node-id=10452-22473&mode=design&t=rel3TgaKShihQBzJ-4)

This will be almost the same as the Inisghts Hub page but with a new component "List Page Card" (similar to [Insight Page Card](https://design-insights.netlify.app/?path=/docs/design-system-molecules-insightpagecard--docs) but for lists) and a new tab component to switch from the Inisghts Hub page to the Lists Hub page.

image

### New Components

- [x] #1668

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### LINKED ISSUE #1668: Feature: Page Tabs for Insights/Lists Hub
### Type of feature

🍕 Feature

### Current behavior

N/A

### [Figma Link](https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/Design-Lab?type=design&node-id=10472-25452&mode=design&t=rel3TgaKShihQBzJ-4)

image

image

Inactive [ .text-light-slate-8 .font-semibold .text-3xl ]
Active [ .text-light-slate-12 .font-semibold .text-3xl .pr-3 ]
Title [  .text-sauced-orange .text-normal .font-semibold ] 

image

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR:
- fixes the alignment of the CTA button in the list (all thanks to @brandonroberts and @jpmcb for noticing the miss alignment in https://github.com/open-sauced/app/pull/1687)
- adds a type for list contributors and a refactor to replace dummy contributors with live data
- adds fetch hooks for lists and contributors
-  refactors the code to break out pages and adds a loader
- adds a redirect to insights tab when a user visits `/hub` route by default. (This is subject to change in the future)

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [x] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #1668 
Closes #1666

## Mobile & Desktop Screenshots/Recordings
image

### No list
image

### With list
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: pages/hub/insights/index.tsx [hunk 1]
```diff
@@ -16,14 +17,27 @@ import useUserInsights from "lib/hooks/useUserInsights";
...
+import ListCard from "components/molecules/ListCard/list-card";
+import { useFetchAllLists } from "lib/hooks/useList";
+import SkeletonWrapper from "components/atoms/SkeletonLoader/skeleton-wrapper";
+
+type TabKey = "insights" | "lists";
+
+const tabs: Record<TabKey, string> = {
+  insights: "Insights",
+  lists: "Lists",
+};
```
Comment (jpmcb) [inline pages/hub/insights/index.tsx:30]: These tabs don't seem to be getting the right styling for me 🤔 (on either firefox or chrome)

Screenshot 2023-09-11 at 8 29 32 AM

This should be more "greyed out"?

File: pages/hub/insights/index.tsx [hunk 2]
```diff
@@ -42,69 +56,155 @@ const InsightsHub: WithPageLayout = () => {
...
+        <TabsContent value={"lists" satisfies TabKey}>
+          <ListTab />
+        </TabsContent>
+
+        <Link
+          passHref
+          href={"/hub/insights/new"}
+          className="w-full py-5 text-lg text-center border rounded-lg bg-light-slate-4 text-light-slate-11 md:py-8 lg:py-10 border-light-slate-7"
+        >
+          {router.pathname.split("/")[2] === "insights" ? "Create a new Insight Page" : "Create a new List"}
```
Comment (jpmcb) [inline pages/hub/insights/index.tsx:113]: This should route to the `NewListCreationPage`? When I click this, it sends me to the new insights page

File: components/molecules/ListCard/list-card.tsx [hunk 3]
```diff
@@ -67,15 +55,14 @@ const ListCard = ({ list }: ListCardProps) => {
...
+              <button className="inline-block p-3 mr-2 border rounded-lg cursor-pointer bg-light-slate-1" type="button">
+                <RiDeleteBinLine title="Edit Insight Page" className="text-lg text-light-slate-10" />
               </button>
-              <Link href={`/list/${list.user.login}/${list.id}/overview`}>
-                <span className=" bg-light-slate-1 inline-block rounded-lg p-2.5 border cursor-pointer">
-                  <MdOutlineArrowForwardIos title="Go To Insight Page" className="text-lg text-light-slate-10" />
-                </span>
+              <Link
+                className="inline-block p-3 mr-2 border rounded-lg cursor-pointer bg-light-slate-1"
+                href={`/list/${list.user.login}/${list.id}/overview`}
```
Comment (brandonroberts) [inline components/molecules/ListCard/list-card.tsx:63]:
```suggestion
                href={`/lists/${list.id}/overview`}
```

File: pages/hub/insights/index.tsx [hunk 4]
```diff
@@ -42,69 +56,155 @@ const InsightsHub: WithPageLayout = () => {
...
-          <Text className="my-8">
-            Welcome to your Insights Hub! Here, you can set up pages to view all of your insights or other open source
-            insights in one place.
-          </Text>
-        </div>
-        <div className="flex items-center gap-3 mt-4">
-          {/* Search box temporarily hidden */}
-          <div className="hidden w-58">
-            <Search placeholder="Search repositories" className="max-w-full" name={"query"} />
+    <Tabs value={router.pathname.split("/")[2] as TabKey} onValueChange={onTabChange}>
```
Comment (brandonroberts) [inline pages/hub/insights/index.tsx:66]: Why can't these be do distinct pages/components? I think the header should be common between the two in the layout, but they don't need to be tabs

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: pages/hub/insights/index.tsx (hunk 4 above)
Comment (OgDev-01) [reply pages/hub/insights/index.tsx]: Thanks for this suggestion. I've implemented in the latest changes 👍

File: pages/hub/insights/index.tsx (hunk 2 above)
Comment (OgDev-01) [reply pages/hub/insights/index.tsx]: Oops... missed that 😄.

updated now to the correct link, but the actual page will be created at https://github.com/open-sauced/app/pull/1698

**Review by brandonroberts — APPROVED** [review]
LGTM 🍕

**Review by jpmcb — APPROVED** [review]
Very nice very nice!! 🥇

**Pull request merged**